pyrit/prompt_target/openai/openai_response_target.py

```
@@ -90,6 +99,10 @@ def __init__(
                 this target with different models, is_json_supported should be set correctly to avoid issues when
                 using adversarial infrastructure (e.g. Crescendo scorers will set this flag).
             extra_body_parameters (dict, Optional): Additional parameters to be included in the request body.
+            fail_on_missing_function: if True, raise when a function_call references
```

why is this configurable ? I feel like it should always fail if the user specifies an unknown function call ? Picking another tool seems misleading and isn't asking for clarification kind of the same as raising an error

---

it's not if the user specifies an unknown function call but rather when the model outputs a function that wasn't specified in the list of custom functions provided by the user (or no function). When this flag is True, this passes a custom formatted error to the model to allows it to try again (with the cost of extra calls to the endpoint and time so I think configurable makes sense). 

In the responses API docs, OpenAI says "A result must be a string, but the format is up to you (JSON, error codes, plain text, etc.). The model will interpret that string as needed." so these types of error codes can be interpreted by the model as a signal to try again with the function list provided.